CHANGELOG.md

```diff
@@ -1,3 +1,259 @@
+  ```sh
+  export PLATFORM="packet" && export ASSETS_DIR="assets"
+  ``` 
+
+3. Remove old asset files:
+  ```sh
+  rm $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/require.tf \
+  $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/workers/require.tf \
+  $ASSETS_DIR/terraform-modules/dns/route53/require.tf 2> /dev/null
+  ```
+
+4. Go to terraform directory:
+  ```sh
+  cd $ASSETS_DIR/terraform
+  ```
+
+5. Replace old providers:
+  ```sh
+  terraform state replace-provider -auto-approve registry.terraform.io/-/ct registry.terraform.io/poseidon/ct && \
+  terraform state replace-provider -auto-approve registry.terraform.io/-/template registry.terraform.io/hashicorp/template
+  ```
+
+6. **NOTE:** delete old ReplicaSet `kube-apiserver` only for single node cluster.
+  ```sh
+  kubectl delete rs $(kubectl get rs -n kube-system -l k8s-app=kube-apiserver --no-headers=true  | sort -k5 -h | head -n +2 | awk '{print $1}')
+  ```
+
+7. FelixConfiguration to be managed by Helm:
+  ```sh
+  kubectl label FelixConfiguration default app.kubernetes.io/managed-by=Helm && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-name=calico && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-namespace=kube-system
+  ```
+
+Finally, run:
+
+```sh
+lokoctl cluster apply --skip-components -v
+```
+
+The update process typically takes about 10 minutes.
+After the update, running `lokoctl health` should result in an output similar to the following:
+
+```sh
+Node                     Ready    Reason          Message
+
+lokomotive-controller-0  True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-0    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-1    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-2    True     KubeletReady    kubelet is posting ready status
+Name      Status    Message              Error
+
+etcd-0    True      {"health":"true"}
+```
+
+#### Cluster nodes component upgrade (optional)
+
+- Manually upgrade etcd following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-etcd.md).
+- Manually upgrade the kubelet running on the nodes, by following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-bootstrap-kubelet.md).
+
+#### cert-manager Upgrade steps
+
+These steps are curated from the upgrade doc provided by [cert-manager](https://cert-manager.io/docs/installation/upgrading/upgrading-0.16-1.0/#kubernetes-1-16-and-above):
+
+Apply manifest:
+
+  ```sh
+  kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.3/cert-manager.yaml
+  ```
+
+Now it is safe to update:
+
+```
+lokoctl component apply cert-manager
+```
+
+#### Upgrade other components
+
+Other components are safe to upgrade by running the following command:
```

Comment:
I tried to upgrade a cluster with Prometheus Operator and it got stuck, `kubectl get events` shows this:

```
98s         Warning   FailedCreate                   job/prometheus-operator-admission-create                                            Error creating: pods "prometheus-operator-admission-create-rtc87" is forbidden: PodSecurityPolicy: unable to validate pod: [pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/create]: Forbidden: runtime/default is not an allowed seccomp profile. Valid values are docker/default spec.securityContext.fsGroup: Invalid value: []int64{0}: group 0 must be in the ranges: [{1 65535}] spec.securityContext.supplementalGroups: Invalid value: []int64{0}: group 0 must be in the ranges: [{1 65535}]]
```

I remember encountering this when I did an upgrade test on one of my clusters with @invidian. @invidian do you remember what the cause was and how we got past that?

Reply:
During offline discussion. We have to delete the `psp prometheus-operator-admission` as it does not apply annotation `runtime/default` while updating. Opened an issue for it https://github.com/kinvolk/lokomotive/issues/1128